karma.conf.js

```diff
@@ -1,128 +1,43 @@
-            require('karma-firefox-launcher'),
-            require('karma-ie-launcher'),
-            require('karma-sinon-chai'),
-            require('karma-coverage'),
-            require('karma-spec-reporter'),
-            require('karma-sourcemap-loader')
-        ],
-
-        webpackMiddleware: {
-            noInfo: true,
-            stats: false
-        },
-
-        webpack: {
-            devtool: 'inline-source-map',
-
-            resolve: {
-                modules: [
-                    __dirname,
-                    'node_modules'
-                ]
-            },
-
-            module: {
-                rules: [
-                    {
-                        test: /\.js$/,
-                        exclude: /(dist|chai)/,
-                        loader: 'babel-loader',
-                        query: {
-                            presets: ['es2015'],
-                            plugins: [
-                                'transform-flow-strip-types',
-                                'transform-es3-property-literals',
-                                'transform-es3-member-expression-literals',
-                                ['transform-es2015-for-of', {loose: true}],
-                                ['flow-runtime', {
-                                    'assert': true,
-                                    'annotate': true
-                                }],
-                                [ '__coverage__', { only: `${__dirname}/src` } ]
-                            ]
-                        }
-                    },
-                    {
-                        test: /\.(html?|css|json)$/,
-                        loader: 'raw-loader'
-                    }
-                ]
-            },
-            bail: false,
-            plugins: [
-                new webpack.DefinePlugin({
-                    __TEST__: true
-                })
-            ]
+        preprocessors: {
+            'test/index.js': 'webpack'
         },
 
+        webpack: testConfig,
 
+        
         // list of files to exclude
         exclude: [
         ],
 
-
-        // preprocess matching files before serving them to the browser
-        // available preprocessors: https://npmjs.org/browse/keyword/karma-preprocessor
-        preprocessors: {
-            'test/test.js': ['webpack', 'sourcemap'],
-            'test/windows/**/*.js': ['webpack', 'sourcemap'],
-            'src/**/*.js': ['coverage',  'sourcemap']
-        },
-
         // test results reporter to use
         // possible values: 'dots', 'progress'
         // available reporters: https://npmjs.org/browse/keyword/karma-reporter
-        reporters: [
-            'spec',
-            'coverage'
-        ],
-
-        coverageReporter: {
-
-            instrumenterOptions: {
-                istanbul: { noCompact: true }
-            },
-
-            reporters: [
-                {
-                    type: 'text'
-                },
-                {
```

Comment:
Right -- but without that `subdir` option they go into a directory like `coverage/PhantomJS 2.1.1 (Mac OS X 0.0.0)/` which isn't great if we ever want to do anything like scripting against the coverage outputs. Can you please add that option back?

Just for future -- I appreciate all of the work put into this, but could you keep cleanup to a separate PR in future? It makes it a lot easier to review if a PR represents the minimal set of changes needed to do something like upgrade to npm build scripts. And it's burning a lot of time reviewing all of this cleanup, since a lot of the stuff that was removed was there for a good reason in the first place.